packages/core/src/ElementOption.ts

```diff
@@ -0,0 +1,171 @@
+import { IReporter, VerboseReporter, BaseReporter, ReportCache } from '@flood/element-report'
+import { TestCommander } from './Runner'
+import { FloodProcessEnv, RuntimeEnvironment } from './runtime-environment/types'
+import WorkRoot from './runtime-environment/WorkRoot'
+import { TestSettings } from './runtime/Settings'
+import { TestObserver } from './runtime/test-observers/TestObserver'
+import { AsyncFactory } from './utils/Factory'
+import { watch } from 'chokidar'
+import { EventEmitter } from 'events'
+import { extname, basename, join, dirname, resolve } from 'path'
+import sanitize from 'sanitize-filename'
+import { PlaywrightClient } from './driver/Playwright'
+import { BROWSER_TYPE } from './page/types'
+
+export interface ElementRunArguments {
+	testFiles: string[]
+	file: string
+	chrome?: string
+	strict?: boolean
+	headless?: boolean
+	devtools?: boolean
+	sandbox?: boolean
+	loopCount?: number
+	stepDelay?: number
+	actionDelay?: number
+	fastForward?: boolean
+	slowMo?: boolean
+	watch?: boolean
+	'work-root'?: string
+	'test-data-root'?: string
+	'fail-status-code': number
+	configFile: string
+	verbose?: boolean
+	browserType: BROWSER_TYPE
+}
+
+export interface ElementOptions {
+	runEnv: RuntimeEnvironment
+	reporter: IReporter
+	clientFactory?: AsyncFactory<PlaywrightClient>
+	testScript: string
+	strictCompilation: boolean
+	headless: boolean
+	devtools: boolean
+	sandbox: boolean
+	process?: NodeJS.Process
+	verbose: boolean
+	testSettingOverrides: TestSettings
+	testObserverFactory?: (t: TestObserver) => TestObserver
+	persistentRunner: boolean
+	testCommander?: TestCommander
+	failStatusCode: number
+	cache?: ReportCache
+	browserType: BROWSER_TYPE
+}
+
+function getWorkRootPath(file: string, root?: string): string {
+	const ext = extname(file)
+	const bare = basename(file, ext)
+
+	if (root == null) {
+		root = join(dirname(file), 'tmp', 'element-results', bare)
+	}
+
+	const dateString = sanitize(new Date().toISOString())
+
+	return resolve(root, dateString)
+}
+
+function getTestDataPath(file: string, root?: string): string {
+	root = root || dirname(file)
+	return resolve(root)
+}
+
+function setupDelayOverrides(
+	args: ElementRunArguments,
+	testSettingOverrides: TestSettings,
+): TestSettings {
+	if (testSettingOverrides == null) testSettingOverrides = {}
+	const { actionDelay, stepDelay } = args
+
+	testSettingOverrides.actionDelay = actionDelay && actionDelay > 0 ? actionDelay : 0
+	testSettingOverrides.stepDelay = stepDelay && stepDelay > 0 ? stepDelay : 0
```

Comment:
If there are no args `actionDelay` and `stepDelay` from the terminal, we do not initialize in the testSettingOverrides, it will override the default setting.

```suggestion
	if (actionDelay) {
		if (typeof actionDelay === 'string') {
			actionDelay = ms(actionDelay)
		}
		testSettingOverrides.actionDelay = actionDelay
	}

	if (stepDelay) {
		if (typeof stepDelay === 'string') {
			stepDelay = ms(stepDelay)
		}
		testSettingOverrides.stepDelay = stepDelay
	}
```